lib/view_model/widgets_view_models/custom_drawer_view_model.dart

```diff
@@ -30,16 +31,27 @@ class CustomDrawerViewModel extends BaseModel {
   }
 
   switchOrg(OrgInfo switchToOrg) {
-    if (selectedOrg == switchToOrg) {
+    if (selectedOrg == switchToOrg && isPresentinSwitchableOrg(switchToOrg)) {
```

Comment:
Can you elaborate on this why is this additional check added. Since the drawer shows the list of only joined orgs present in the list and the function takes in the input param of the org on which the user tap then in no such case the org selected is going to be absent in the list of joined orgs.

Reply:
Yeah so this for testing ,if in the future someone updates the business logic such that it's faulty and a org is selected which is not present in the switchable then switching of org should not happen.

Reply:
And also while unit testing we should be testing the function fully ,independent of the previous business logic attached with it.

Reply:
Can you please elaborate on this? My point is since the selected org is a stream then in what case can the organization getting selected wouldn't be present in the joined organization list.

Reply:
Yeah so basically there is a testing in the test file as-
```
Test('check if switchOrg is working with wrong switchable org being passed',
        () async {
      model.setSelectedOrganizationName(userConfig.currentOrg);

      //Mock switchable org are present in the model
      model.switchAbleOrg = userConfig.currentUser.joinedOrganizations!;

      //Mock fake org which is not present in the mock switchableOrg

      //Acess mock joined Organisation for the mock user
      final OrgInfo fakeOrg = OrgInfo(
        id: '5',
        name: 'fake org 1',
        isPublic: false,
        creatorInfo: User(firstName: 'fake', lastName: 'user'),
      );
      //check if the mocked org is present or not

      final isPresent = model.isPresentinSwitchableOrg(fakeOrg);

      //expecting that the org is not present so will return false
      expect(isPresent, false);
      //check if selected org is changed or not. Expected-Not changing
      expect(model.selectedOrg, isNot(fakeOrg));
    });
```

So this test the switchOrg function individually irrespective of the previous checks done on other function.But if you want i can remove and recommit.